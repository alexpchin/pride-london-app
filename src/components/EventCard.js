// @flow
import React from "react";
import { View, StyleSheet, ImageBackground } from "react-native";
import formatDate from "date-fns/format";
import {
  eventListBgColor,
  priceTagBgColor,
  imageBgColor,
  cardBgColor,
  textColor
} from "../constants/colors";
import Text from "./Text";

type Props = {
  name: string,
  locationName: string,
  price: number,
  startTime: string,
  endTime: string,
  imageUrl: string
};
const removeTimezoneFromDateString = isoString => isoString.slice(0, -6);

const EventCard = ({
  name,
  locationName,
  startTime,
  endTime,
  price,
  imageUrl
}: Props) => {
  const eventStartTime = removeTimezoneFromDateString(startTime);
  const eventEndTime = removeTimezoneFromDateString(endTime);
  const timeFormat = "HH:mm";
  const timeDisplay = `${formatDate(eventStartTime, timeFormat)} - ${formatDate(
    eventEndTime,
    timeFormat
  )}`;
  return (
    <View style={styles.eventCard}>
      <ImageBackground
        style={styles.imageContainer}
        source={{ uri: imageUrl }}
        resizeMode="cover"
      >
        <View style={styles.eventPriceContainer}>
          <Text style={styles.eventPrice}>Â£{price}</Text>
        </View>
      </ImageBackground>
      <View style={styles.eventCardDetails}>
        <Text style={styles.eventTime}>{timeDisplay}</Text>
        <View style={styles.eventNameContainer}>
          <Text style={styles.eventName}>{name}</Text>
        </View>
        <Text style={styles.eventLocation}>{locationName}</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  eventCard: {
    height: 108,
    backgroundColor: cardBgColor,
    flexDirection: "row",
    overflow: "hidden"
  },
  imageContainer: {
    width: 114,
    height: 108,
    backgroundColor: imageBgColor
  },
  eventPriceContainer: {
    height: 23,
    backgroundColor: priceTagBgColor,
    position: "absolute",
    paddingHorizontal: 8,
    borderRadius: 2
  },
  eventCardDetails: {
    flex: 1,
    padding: 8
  },
  eventNameContainer: {
    flexDirection: "row"
  },
  eventName: {
    fontFamily: "Poppins-SemiBold",
    color: textColor,
    fontSize: 16,
    lineHeight: 20,
    paddingTop: 4
  },
  eventPrice: {
    fontSize: 14,
    color: eventListBgColor
  },
  eventTime: {
    fontSize: 14,
    lineHeight: 20
  },
  eventLocation: {
    fontSize: 12,
    lineHeight: 16,
    paddingTop: 4
  }
});

export default EventCard;
