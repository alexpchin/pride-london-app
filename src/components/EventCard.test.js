// @flow
import React from "react";
import { shallow } from "enzyme";
import EventCard from "./EventCard";

it("renders correctly", () => {
  const output = shallow(
    <EventCard
      name="name"
      locationName="location"
      price={16}
      startTime="2018-09-16T12:12:12+01:00"
      endTime="2018-09-16T13:12:12+01:00"
      imageUrl="https://image.jpg"
    />
  );
  expect(output).toMatchSnapshot();
});
