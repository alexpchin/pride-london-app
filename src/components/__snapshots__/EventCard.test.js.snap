// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flexDirection": "row",
      "height": 108,
      "overflow": "hidden",
    }
  }
>
  <ImageBackground
    resizeMode="cover"
    source={
      Object {
        "uri": "https://image.jpg",
      }
    }
    style={
      Object {
        "backgroundColor": "#C1C1C1",
        "height": 108,
        "width": 114,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#989898",
          "borderRadius": 2,
          "height": 23,
          "paddingHorizontal": 8,
          "position": "absolute",
        }
      }
    >
      <Text
        markdown={false}
        style={
          Object {
            "color": "rgb(226, 226, 226)",
            "fontSize": 14,
          }
        }
        type="text"
      >
        Â£
        16
      </Text>
    </View>
  </ImageBackground>
  <View
    style={
      Object {
        "flex": 1,
        "padding": 8,
      }
    }
  >
    <Text
      markdown={false}
      style={
        Object {
          "fontSize": 14,
          "lineHeight": 20,
        }
      }
      type="text"
    >
      12:12 - 13:12
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        markdown={false}
        style={
          Object {
            "color": "#272727",
            "fontFamily": "Poppins-SemiBold",
            "fontSize": 16,
            "lineHeight": 20,
            "paddingTop": 4,
          }
        }
        type="text"
      >
        name
      </Text>
    </View>
    <Text
      markdown={false}
      style={
        Object {
          "fontSize": 12,
          "lineHeight": 16,
          "paddingTop": 4,
        }
      }
      type="text"
    >
      location
    </Text>
  </View>
</View>
`;
